### PR TITLE
Add 'useEmail' metrics event

### LINKED ISSUE #9834: [Bug] It seems like we don't track the metrics of the "Use Email" button
It seems like we don't track the metrics of the "Use Email" button. It was removed in https://github.com/mozilla-mobile/fenix/pull/4931/files#diff-c0328a8e65981c7ed9bdbb1d366c2849R225

### Expected behavior

Being able to tell when users press the `use email` button

### Actual behavior

We can only see successful sign_in flows after they happen

### Device information

* Android device: Pixel 5
* Fenix version: master

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #9834

Fixes a metrics regression introduced in https://github.com/mozilla-mobile/fenix/pull/4931/files#diff-c0328a8e65981c7ed9bdbb1d366c2849R225 with the removal of the event.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
   **explanation**: telemetry changes, didn't see any tests for this particular code path.
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
  **explanation**: no visual changes
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/metrics.yaml [hunk 1]
```diff
@@ -841,6 +841,17 @@ sync_auth:
...
     - fenix-core@mozilla.com
     expires: "2020-09-01"
+  use_email:
+    type: event
+    description: >
+      A user chose to use their email to sign in instead of scanning a QR code
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/9834
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/issues/9835
```
Comment (grigoryk) [inline app/metrics.yaml:851]: This needs to be a link to a data review response (once you get it).

File: app/metrics.yaml [hunk 2]
```diff
@@ -841,6 +841,17 @@ sync_auth:
     notification_emails:
     - fenix-core@mozilla.com
     expires: "2020-09-01"
+  use_email:
+    type: event
+    description: >
+      A user chose to use their email to sign in instead of scanning a QR code
```
Comment (grigoryk) [inline app/metrics.yaml:847]: Maybe mention the pairing counter-part of this event? I'm guessing it's `paired`, which is a little confusing - that name implies that `paired` is logged after user successfully signed-in via pairing. This event, I'm assuming, is emitted when user begins to login via email. So, they're not 1-1 comparable? Or is it just bad naming?

Comment (vladikoff) [reply app/metrics.yaml:847]: The counter part is `scan_pairing`.

`paired` counterparts are `sign_in` , `sign_up` or external, etc

**Comment by vladikoff** [comment]
cc [@boek](https://github.com/boek) [@liuche](https://github.com/liuche) as mobile data stewards.

Request for data collection review form
=======================================

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1.  What questions will you answer with this data?

How users are interacting with FxA in Fenix - creating an account, login. Specifically when the users choose to use the email login over other means.

2.  Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:

We need to provide our users with a seamless, robust and easy-to-use FxA experience, and these metrics help us understand if our implementation is working as expected, and which kinds of authentication flows users are following.

3.  What alternative methods did you consider to answer these questions? Why were they not sufficient?

An alternative could be to rely FxA metrics, but these do not allow us to complete a coherent picture of what's going on in a local authentication funnel, and some of the key local events (account recovery, auto-login) are not visible outside of the local client.

4.  Can current instrumentation answer these questions?

No. We tried to measure this and found this regression.

5.  List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) on the Mozilla wiki.

All data is Category 2.

6.  How long will this data be collected? Choose one of the following:

This is scoped to a time-limited experiment/project until date "2020-09-01", at which point it'll be reviewed for renewal.

7.  What populations will you measure?

All release, beta, and nightly users with telemetry enabled.

8.  If this data collection is default on, what is the opt-out mechanism for users?

This is part of events telemetry, so can be disabled via Data Collection settings.

9.  Please provide a general description of how you will analyze this data.

Dashboards in redash and amplitude from glean data.

10.  Where do you intend to share the results of your analysis?

Internally with the mobile teams, and on redash/aplitude.

11.  Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection? If so:

N/A

File: app/metrics.yaml [hunk 3]
```diff
@@ -841,6 +841,28 @@ sync_auth:
...
+      - https://github.com/mozilla-mobile/fenix/issues/9834
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/pull/9835#issuecomment-612951889
+    notification_emails:
+      - fenix-core@mozilla.com
+    expires: "2020-09-01"
+  use_email_problem:
+    type: event
+    description: >
+      A user chose to use their email to sign after an account problem
```
Comment (grigoryk) [inline app/metrics.yaml:858]: spelling: to *sign **in***

File: app/metrics.yaml [hunk 4]
```diff
@@ -841,6 +841,28 @@ sync_auth:
...
     - fenix-core@mozilla.com
     expires: "2020-09-01"
+  use_email:
+    type: event
+    description: >
+      A user chose to use their email to sign in instead of scanning a QR code, counterpart to "scan_pairing"
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/9834
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/pull/9835#issuecomment-612951889
```
Comment (grigoryk) [inline app/metrics.yaml:851]: This should be a link to the data steward's response, not the review form itself.

Comment (ekager) [reply app/metrics.yaml:851]: Ideally, but sometimes we also just link to the PR with both of the forms 🤷‍♀️

Comment (vladikoff) [reply app/metrics.yaml:851]: I was just following the best example I had https://github.com/mozilla-mobile/fenix/pull/4931#issuecomment-529740300

Comment (grigoryk) [reply app/metrics.yaml:851]: Every time I've had a review from from glean folks, they've been pretty insistent on it :)

**Comment by vladikoff** [comment]
Rebased and resolved the metrics doc comment 👍

**Comment by vladikoff** [comment]
Rebased and updated with the review link. Thank you Jeff!

**Pull request merged**